File: backend/src/JSO.Domain/MediaAsset.cs
namespace JSO.Domain;

public sealed class MediaAsset
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Title { get; set; } = null!;
    public string Url { get; set; } = null!;
    public string Type { get; set; } = "Image";
    public string? ThumbnailUrl { get; set; }
    public string? Caption { get; set; }
    public bool IsPublished { get; set; } = true;
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public string? FileName { get; set; }
    public string? ContentType { get; set; }
    public long? FileSize { get; set; }
    public string? StoragePath { get; set; }
}

File: mobile/lib/data/models/media_asset.dart
import 'json_utils.dart';

/// Media asset — maps `JSO.Domain.MediaAsset`.
class MediaAsset {
  const MediaAsset({
    required this.id,
    required this.title,
    required this.url,
    required this.type,
    this.thumbnailUrl,
    this.caption,
    required this.createdAt,
  });

  final String id;
  final String title;
  final String url;

  /// e.g. "Image" or "Video".
  final String type;
  final String? thumbnailUrl;
  final String? caption;
  final DateTime createdAt;

  factory MediaAsset.fromJson(Map<String, dynamic> json) => MediaAsset(
    id: asString(json['id']),
    title: asString(json['title']),
    url: asString(json['url']),
    type: asString(json['type'], fallback: 'Image'),
    thumbnailUrl: asStringOrNull(json['thumbnailUrl']),
    caption: asStringOrNull(json['caption']),
    createdAt: asDateTime(json['createdAt']),
  );
}
